spec/features/users/user_updates_profile_spec.rb

```diff
@@ -7,11 +7,12 @@
 
     visit edit_users_path
 
-    fill_in "user_email", with: "JT@example.com"
-    fill_in "user_name", with: user.name
+    fill_in "user_name", with: "Fred"
+    fill_in "user_email", with: user.email
+    fill_in "user_password", with: user.password
```

Comment:
Do these two fields need to be filled manually? can the form work without them?

Websites usually have a different form/workflow for password changing. Maybe think about what the user experience you wish to create.

And a side note: implement reseting your password, if the user cannot log in (in a future PR)

Reply:
The email one didn't, it is populated when you arrive at the form but the password isn't?
I will look at password reset and changing in a later PR.
Done in sha: 1159cb8